swap/tests/happy_path.rs

```diff
@@ -94,8 +92,8 @@ async fn happy_path() {
         Uuid::new_v4(),
     );
 
-    let _bob_swarm_fut = tokio::spawn(async move { bob_event_loop.run().await });
-
+    tokio::spawn(async move { alice_event_loop.run().await });
+    tokio::spawn(async move { bob_event_loop.run().await });
```

Comment:
Could the stackoverflow problem be related to removing the futures being kept in scope with `_` variables?
I don't think it should, but that is the only notable change I can see that could cause this.

Reply:
The  `JoinHandle` returned by `tokio::spawn` is only meant to be used to await on the stack if one wishes to do so.